### PR TITLE
Return importanceRank along with program attachments

### PR DESCRIPTION
**Pull request by rjgildea**

### CODE REVIEW

**Review by KarlLevik — CHANGES_REQUESTED** [review]
I haven't tested these, but I'm fairly certain that the SPs containing aggregate functions (concat, group_concat, ...) and `GROUP BY` will need the appa.importanceRank column in the `SELECT` clause moved to be inside the aggregate function. Just add it as a field in the json_object, similar to the other appa.* columns.

**Comment by rjgildea** [comment]
Thanks - I hadn't looked closely enough to understand exactly what the SP was doing/returning. Now included in the `json_object` call.

File: schema/stored_programs/sp_retrieve_processing_program_attachments_for_dc_group_program_v2.sql [hunk 1]
```diff
@@ -14,7 +14,8 @@ BEGIN
         IF p_authLogin IS NOT NULL THEN
             SELECT dc.dataCollectionId, app.autoProcProgramId,
                 app.processingStatus,
-                concat('[', group_concat(json_object('fileType', appa.fileType, 'fullFilePath', concat(appa.filePath, '/', appa.fileName))), ']') "processingAttachments"
+                concat('[', group_concat(json_object('fileType', appa.fileType, 'fullFilePath', concat(appa.filePath, '/', appa.fileName), 'importanceRank', appa.importanceRank)), ']') "processingAttachments"
+                appa.importanceRank
```
Comment (KarlLevik) [inline schema/stored_programs/sp_retrieve_processing_program_attachments_for_dc_group_program_v2.sql:18]: I think we want to remove your L18.

File: schema/routines.sql [hunk 2]
```diff
@@ -3883,7 +3883,7 @@ BEGIN
     IF p_id IS NOT NULL AND p_program IS NOT NULL THEN
       SELECT dc.dataCollectionId, app.autoProcProgramId,
         app.processingStatus,
-        concat('[', group_concat(json_object('fileType', appa.fileType, 'fullFilePath', concat(appa.filePath, '/', appa.fileName))), ']') "processingAttachments"
+        concat('[', group_concat(json_object('fileType', appa.fileType, 'fullFilePath', concat(appa.filePath, '/', appa.fileName, 'importanceRank', appa.importanceRank))), ']') "processingAttachments"
```
Comment (KarlLevik) [inline schema/routines.sql:3886]: The `importanceRank` field should not be concat'ed in the `fullFilePath`! It should be outside of that. Looks like you got it correctly in the `sp_retrieve_processing_program_attachments_for_dc_group_and_program.sql` file below.

**Review by KarlLevik — CHANGES_REQUESTED** [review]
Found two issues, see inline comments. Looks good other than that.

File: schema/routines.sql (hunk 2 above)
Comment (rjgildea) [reply schema/routines.sql]: Too many parentheses!

**Review by KarlLevik — APPROVED** [review]

**Pull request merged**